Dockerfile

```diff
@@ -1,80 +1,118 @@
 ARG BUILD_BASE=ubuntu:${UBUNTU_VERSION}
-ARG LIBRARIES=/opt/trailofbits/libraries
+ARG LIBRARIES=/opt/trailofbits
+ARG BINJA_DECODE_KEY
 
-# Will copy remill installation from here
-FROM trailofbits/remill:llvm${LLVM_VERSION}-${DISTRO_BASE}-${ARCH} as remill
-
-# Additional runtime dependencies go here
-FROM ${BUILD_BASE} as base
+# Run-time dependencies go here
+FROM ${BUILD_BASE} AS base
 ARG UBUNTU_VERSION
+ARG LIBRARIES
 RUN apt-get update && \
-    cat /etc/lsb-release && \
-    if [ "${UBUNTU_VERSION}" = "20.04" ] ; then \
-        apt-get install -qqy --no-install-recommends libtinfo6 ; \
-    else \
-        apt-get install -qqy --no-install-recommends libtinfo5 ; \
-    fi && \
+    apt-get install -qqy --no-install-recommends python3 python3-pip python3.8 python3.8-venv python3-setuptools xz-utils && \
     rm -rf /var/lib/apt/lists/*
 
 # Build-time dependencies go here
-FROM trailofbits/cxx-common:llvm${LLVM_VERSION}-${DISTRO_BASE}-${ARCH} as deps
-
-ENV DEBIAN_FRONTEND=noninteractive
+FROM trailofbits/cxx-common-vcpkg-builder-ubuntu:${UBUNTU_VERSION} as deps
+ARG UBUNTU_VERSION
+ARG ARCH
+ARG LLVM_VERSION
+ARG LIBRARIES
 
 RUN apt-get update && \
-    apt-get install -qqy ninja-build python3.8 python3-pip python3.8-venv liblzma-dev zlib1g-dev libtinfo-dev curl git wget build-essential ninja-build ccache clang && \
+    apt-get install -qqy xz-utils python3.8-venv make rpm && \
     rm -rf /var/lib/apt/lists/*
 
-COPY --from=remill /opt/trailofbits /opt/trailofbits
+# Build dependencies
+WORKDIR /dependencies
+RUN git clone --depth=1 --branch master https://github.com/lifting-bits/remill.git && \
+    cd remill && \
+    ./scripts/build.sh --llvm-version ${LLVM_VERSION} --prefix ${LIBRARIES} --download-dir /dependencies
 
-# Source code build
-FROM deps as build
-ARG LIBRARIES
+# Make this a separate RUN because the build script above downloads a lot
+RUN cd remill && \
+    dpkg -i remill-build/*.deb
 
+# Source code build
+FROM deps AS build
 WORKDIR /anvill
-COPY . ./
+ARG UBUNTU_VERSION
+ARG ARCH
+ARG LLVM_VERSION
+ARG LIBRARIES
 
-ENV CC="/usr/bin/clang"
-ENV CXX="/usr/bin/clang++"
-ENV TRAILOFBITS_LIBRARIES="${LIBRARIES}"
 ENV VIRTUAL_ENV=/opt/trailofbits/venv
-ENV PATH="${VIRTUAL_ENV}/bin:${LIBRARIES}/llvm/bin:${LIBRARIES}/cmake/bin:${LIBRARIES}/protobuf/bin:${PATH}"
+ENV PATH="${VIRTUAL_ENV}/bin:${PATH}"
 
 # create a virtualenv in /opt/trailofbits/venv
 RUN python3.8 -m venv ${VIRTUAL_ENV}
 
-RUN mkdir -p build && cd build && \
-    cmake -G Ninja -DCMAKE_PREFIX_PATH=/opt/trailofbits/remill -DCMAKE_VERBOSE_MAKEFILE=True -DCMAKE_INSTALL_PREFIX=/opt/trailofbits/anvill .. && \
-    cmake --build . --target install
+# Needed for sourcing venv
+SHELL ["/bin/bash", "-c"]
+
+COPY . ./
+
+RUN source ${VIRTUAL_ENV}/bin/activate && \
+    cmake -G Ninja -B build -S . \
+        -DANVILL_ENABLE_INSTALL_TARGET=true \
+        -Dremill_DIR:PATH=${LIBRARIES}/lib/cmake/remill \
+        -DANVILL_INSTALL_PYTHON3_LIBS=ON \
```

Comment:
This is kind of deprecated, as it's not a good practice to change system state during configure or build time. The `ANVILL_ENABLE_INSTALL_TARGET` setting can be used to install to a $DESTDIR, which can be later used to create a package.

Ideally we want to install both the native code and the Python module in the same place (vs using a venv for Python and /usr for the rest). If we want to still support installing inside a venv maybe we can create a .sh installer that we ship in the package

Reply:
I would very much like to support the venv case for our Python code, it was  much easier to integrate with Binja that way. We should be able to get rid of INSTALL_PYTHON3_LIBS with the fixes you merged last night.